src/main/java/io/confluent/connect/hdfs/json/JsonRecordWriterProvider.java

```diff
@@ -0,0 +1,104 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.confluent.connect.hdfs.json;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.hadoop.fs.Path;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import io.confluent.connect.hdfs.HdfsSinkConnectorConfig;
+import io.confluent.connect.hdfs.storage.HdfsStorage;
+import io.confluent.connect.storage.format.RecordWriter;
+import io.confluent.connect.storage.format.RecordWriterProvider;
+
+public class JsonRecordWriterProvider implements RecordWriterProvider<HdfsSinkConnectorConfig> {
+
+  private static final Logger log = LoggerFactory.getLogger(JsonRecordWriterProvider.class);
+  private static final String EXTENSION = ".json";
+  private static final String LINE_SEPARATOR = System.lineSeparator();
+  private static final byte[] LINE_SEPARATOR_BYTES = LINE_SEPARATOR.getBytes();
+  private final HdfsStorage storage;
+  private final ObjectMapper mapper;
+  private final JsonConverter converter;
+
+  JsonRecordWriterProvider(HdfsStorage storage, JsonConverter converter) {
+    this.storage = storage;
+    this.mapper = new ObjectMapper();
+    this.converter = converter;
+  }
+
+  @Override
+  public String getExtension() {
+    return EXTENSION;
+  }
+
+  @Override
+  public RecordWriter getRecordWriter(final HdfsSinkConnectorConfig conf, final String filename) {
+    try {
+      return new RecordWriter() {
+        final Path path = new Path(filename);
+        final OutputStream out = path.getFileSystem(conf.getHadoopConfiguration()).create(path);
+        final JsonGenerator writer = mapper.getFactory()
+            .createGenerator(out)
+            .setRootValueSeparator(null);
+
+        @Override
+        public void write(SinkRecord record) {
+          log.trace("Sink record: {}", record.toString());
+          try {
+            Object value = record.value();
+            if (value instanceof Struct) {
+              byte[] rawJson = converter.fromConnectData(
+                  record.topic(),
+                  record.valueSchema(),
+                  value
+              );
+              out.write(rawJson);
+              out.write(LINE_SEPARATOR_BYTES);
+            } else {
+              writer.writeObject(value);
+              writer.writeRaw(LINE_SEPARATOR);
+            }
+          } catch (IOException e) {
+            throw new ConnectException(e);
+          }
+        }
+
+        @Override
+        public void commit() {}
```

Comment:
Is there any advantage to flushing the output stream with `commit()`? `JsonGenerator` doesn't really seem to offer a flush, but `OutputStream` certainly does.

Reply:
This method is not used in the HDFS connector. It was introduced in the `RecordWriterProvider` interface in `kafka-connect-storage-common` to be used by connectors such as the S3 connector that need to attempt committing offsets before closing the writer.

This connector uses only `close`, which flushes the output stream during `writer.close()` since the JsonGenerator by default manages the underlying stream.

Reply:
Sounds fine. Thanks for the answer.